include/crpropa/PhotonBackground.h

```diff
@@ -2,30 +2,50 @@
 #define CRPROPA_PHOTONBACKGROUND_H
 
 #include "crpropa/Common.h"
+#include "crpropa/Referenced.h"
 
 #include <string>
 
 namespace crpropa {
-
 /**
- * \addtogroup EnergyLosses
+ * \addtogroup PhotonFields
  * @{
  */
 
-
 /**
  @class PhotonField
- @brief photon field class fully defined with a range of photon energies, redshift and the field's density
+ @brief Abstract base class for photon fields.
  */
-class PhotonField {
+class PhotonField: public Referenced {
 public:
-	PhotonField(std::string fieldName, bool hasRedshiftDependence = true);
-	PhotonField() : PhotonField("CMB", false) {};
+	virtual double getPhotonDensity(double ePhoton, double z = 0.) const {
+		return 0.;
+	};
+	virtual double getRedshiftScaling(double z) const {  // returns overall comoving scaling factor (cf. CRPropa3-data/calc_scaling.py)
```

Comment:
by making getPhotonDensity pure virtual, PhotonField has become abstract and would therefore suit our requirement for that class to not be instantiated.
Could one therefore leave getRedshiftScaling semi-virtual since it is always worthwhile for this function to return 1. unless needed otherwise?